docs/design_proposals/configurable-transformable-allowlist.md

```diff
@@ -0,0 +1,91 @@
+# Configurable transformableAllowList
+
+* Author(s): Ke Zhu (@shawnzhu)
+* Design Shepherd: \<skaffold-core-team-member\>
+* Date: 2021-07-16
+* Status: Draft
+
+## Objectives
+
+Configurable transformableAllowList for transforming manifests.
+
+## Background
+
+Skaffold can only transform manifests from a non-extensible allowlist. When
+using any CRD out of this allowlist, skaffold can not transform it.
+
+Open issues concerning this problem:
+
+* Image not recognized in crd k8s manifest ([#4081](https://github.com/GoogleContainerTools/skaffold/issues/4081))
+
+There was comments out of the above issue to make this allowlist extensible.
+
+The goal of this document is to create an agreement on the configuration option
+and specification to extend `skaffold.kubernetes.manifest.transformableAllowList`.
+
+## Design
+
+Add following configuration option:
+
+| Precedence | Kind | Details |
+|------------|----------------------|------------------------------------------|
+| 1.         |`skaffold.yaml`       |Json-path `deploy.transformableAllowList` |
+| 2.         |Global Skaffold config|give users the possibility to extend the default transformableAllowList globally or per project.|
+
+Notice that any new configuration option will be appended to existing allowlist. 
+
+### Detailed discussion
+
+Option in `skaffold.yaml`
+
+```YAML
+deploy:
```

Comment:
I don't have a good suggestion for this. Not a fan of putting this `deploy`  since this is applicable to deploy and debug.
maybe something like `deploySettings` or `config`. I am not sure.